src/mediawiki/templates/InfoboxMusic/InfoboxMusic.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  InfoboxTemplate,
+  MediaWikiDate,
+  MediaWikiFile,
+} from "@osrs-wiki/mediawiki-builder";
+
+import { InfoboxMusic } from "./InfoboxMusic.types";
+
+import Context from "@/context";
+import { DBRow } from "@/utils/cache2";
+
+// Helper function to format duration from seconds to MM:SS
+const formatDuration = (seconds?: number): string | undefined => {
+  if (!seconds || seconds <= 0) return undefined;
+  
+  const minutes = Math.floor(seconds / 60);
+  const secs = seconds % 60;
+  return `${minutes}:${secs.toString().padStart(2, '0')}`;
+};
+
+// Helper function to detect if this is a quest track
+const isQuestTrack = (hint?: string): boolean => {
```

Comment:
Move these to `InfoboxMusic.utils.ts`. Add tests for these util functions in `InfoboxMusic.utils.test.ts`.